pts.go

```diff
@@ -89,27 +89,32 @@ func (p PTS) RolledOver(other PTS) bool {
 	return false
 }
 
-// DurationFrom returns the difference between the two pts times. This number is always positive.
-func (p PTS) DurationFrom(from PTS) uint64 {
-	switch {
-	case p.RolledOver(from):
-		return uint64((MaxPtsTicks - from) + p)
-	case from.RolledOver(p):
-		return uint64((MaxPtsTicks - p) + from)
-	case p < from:
-		return uint64(from - p)
-	default:
-		return uint64(p - from)
+// DurationFrom returns the difference between the two pts times.
+// One possible distance has rollover and wraps around the number line,
+// the other one does not wrap around the number line and doesn't have rollover.
+// DurationFrom will assume the smallest difference is the correct one.
+// This number is always positive.
+// This function will only be accurate if the difference between the two
+// PTS values is less than MaxPtsTicks/2 (2^32 = 4294967296 = 0x100000000)
```

Comment:
I'm not sure about the change to this function.  It operates a bit differently.  It feels like there are some disadvantages in the old way, but different disadvantages in the new way and the new way may be breaking someone relying on the old functionality.

Reply:
They both behave in the same way if the PTS difference is small enough. Thats why it was able to pass the unit tests without removing any test cases. But it will break backwards compatibility if used with large PTS values.

Reply:
The two test cases there `10 - 5`, and `MaxPtsValue - 10 - 5` are pretty basic.

There are a lot of different cases where things behave differently.

Anytime the difference is >= MaxPtsTicks/2 the result will now be different.

I'm not sure this is the correct behavior.  Both the new and old way make assumptions that cannot be known.